pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -605,6 +605,15 @@ func (s *kafkaSink) handleBufferedRetries(msgs []*sarama.ProducerMessage, retryE
 		// SendMessages will attempt to send all messages into an AsyncProducer with
 		// the client's config and then block until the results come in.
 		lastSendErr = newProducer.SendMessages(msgs)
+		if lastSendErr != nil {
+			// nolint:errcmp
+			if sendErrs, ok := lastSendErr.(sarama.ProducerErrors); ok && len(sendErrs) > 0 {
```

Comment:
I was unclear why the linter triggered here and in the mock below as the lint message suggests the format I've done here as an alternative.